src/Microsoft.ML.CodeGen/CodeGenerator/CSharp/CodeGenerator.cs

```diff
@@ -293,6 +322,8 @@ private string GeneratePredictProgramCSFileContent(string namespaceValue)
                 Separator = _columnInferenceResult.TextLoaderOptions.Separators.FirstOrDefault(),
                 AllowQuoting = _columnInferenceResult.TextLoaderOptions.AllowQuoting,
                 AllowSparse = _columnInferenceResult.TextLoaderOptions.AllowSparse,
+                Features = features.Where((str) => str.Name != _settings.LabelName).Select((str)=>str.Name).ToList(),
```

Comment:
Looks to me you're picking features from Columns here, so the lower case features var name is actually deceiving, the Features one will have just features.

Reply:
resolved